fMRIVolume/scripts/IntensityNormalization.sh

```diff
@@ -117,20 +140,49 @@ echo "PWD = `pwd`" >> $WD/log.txt
 echo "date: `date`" >> $WD/log.txt
 echo " " >> $WD/log.txt
 
+########################################## DO WORK ##########################################
 
-########################################## DO WORK ########################################## 
-
-# FinalMask is a combination of the FS-derived brainmask, and the spatial coverage mask that captures the
-# voxels that have data available at *ALL* time points (derived in OneStepResampling)
 FinalMask=`${FSLDIR}/bin/remove_ext ${InputfMRI}`_finalmask
-${FSLDIR}/bin/fslmaths ${BrainMask} -bin -mas ${InputfMRI}_mask ${FinalMask}
+T1FOVMask=`${FSLDIR}/bin/remove_ext ${InputfMRI}`_T1FOVmask
+${FSLDIR}/bin/fslmaths ${BrainMask} -bin -mas ${InputfMRI}_mask ${T1FOVMask}
+
+# Use the requested BOLD mask 
+
+if [ "${BOLDMask}" = "T1_fMRI_FOV" ] ; then
+    # FinalMask is a combination of the FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/fslmaths ${BrainMask} -bin -mas ${InputfMRI}_mask ${FinalMask}
+
+elif [ "${BOLDMask}" = "T1_DILATED_fMRI_FOV" ] ; then
+    # FinalMask is a combination of once dilated FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/fslmaths ${BrainMask} -bin -dilF -mas ${InputfMRI}_mask ${FinalMask}
+
+elif [ "${BOLDMask}" = "T1_DILATED2x_fMRI_FOV" ] ; then
+    # FinalMask is a combination of twice dilated FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/fslmaths ${BrainMask} -bin -dilF -dilF -mas ${InputfMRI}_mask ${FinalMask}
+
+elif [ "${BOLDMask}" = "fMRI_FOV" ] ; then
+    # FinalMask is the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    {FSLDIR}/bin/imcp ${InputfMRI}_mask ${FinalMask}
+
+elif
+    # No valid BOLDMask option was specified
+    log_Err_Abort "Specified BOLD mask to use (--boldmask=${BOLDMask}) is invalid!"
```

Comment:
More direct statement:
`log_Err_Abort "Invalid entry for specified BOLD mask (--boldmask=${BOLDMask})"`

Reply:
changed.